main_test.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"os"
+	"testing"
+
+	"github.com/quux00/ogonori/constants"
+	"github.com/quux00/ogonori/obinary"
+)
+
+var DocumentDBSeeds = []string{
+	"CREATE CLASS Animal",
+	"CREATE property Animal.name string",
+	"CREATE property Animal.age integer",
+	"CREATE CLASS Cat extends Animal",
+	"CREATE property Cat.caretaker string",
+	"INSERT INTO Cat (name, age, caretaker) VALUES ('Linus', 15, 'Michael'), ('Keiko', 10, 'Anna')",
+}
+
+func SetUp(db *obinary.DBClient) {
+	if err := obinary.CreateDatabase(db, dbDocumentName, constants.DocumentDB, constants.Persistent); err != nil {
+		panic(err)
+	}
+	if err := obinary.CreateDatabase(db, dbGraphName, constants.GraphDB, constants.Persistent); err != nil {
+		panic(err)
+	}
+}
+
+func TearDown(db *obinary.DBClient) {
+	obinary.DropDatabase(db, dbDocumentName, constants.DocumentDB)
+	obinary.DropDatabase(db, dbGraphName, constants.GraphDB)
+}
+
+func Seed() {
+	db := ConnectToDocumentDatabase()
+	defer db.Close()
+
+	var err error
+
+	for _, seed := range DocumentDBSeeds {
+		if _, _, err = obinary.SQLCommand(db, seed); err != nil {
+			panic(err) // Programming error
+		}
+	}
+}
+
+func CleanupSeed() {
+	documentDB := ConnectToDocumentDatabase()
+	defer documentDB.Close()
+	obinary.SQLCommand(documentDB, "TRUNCATE Animal")
+	obinary.SQLCommand(documentDB, "TRUNCATE Cat")
+
+	graphDB := ConnectToGraphDatabase()
+	defer graphDB.Close()
+	obinary.SQLCommand(graphDB, "TRUNCATE Person")
+	obinary.SQLCommand(graphDB, "TRUNCATE Friend")
+}
+
+// Runs before all tests
+func TestMain(m *testing.M) {
```

Comment:
testing.M is http://golang.org/pkg/testing/#hdr-Main

> It is sometimes necessary for a test program to do extra setup or teardown before or after testing. It is also sometimes necessary for a test to control which code runs on the main thread. To support these and other cases, if a test file contains a function:

1. TearDown - drop database [this is repeated because sometime locally test fail and seeded data persists]
2. SetUp - create database
3. Seed database
4. Run tests
5. Tear down database